packages/slider/src/useSlider.ts

```diff
@@ -0,0 +1,292 @@
+
+  const init = (initMinValue = min, initMaxValue = max) => {
+    const retVal = {
+      minValue: initMinValue < min ? min : initMinValue,
+      maxValue: initMaxValue > max ? max : initMaxValue
+    };
+
+    if (retVal.maxValue < min) {
+      retVal.maxValue = min;
+    }
+
+    if (retVal.minValue > retVal.maxValue) {
+      retVal.minValue = retVal.maxValue;
+    }
+
+    return retVal;
+  };
+
+  const [state, setState] = useState(init(defaultMinValue, defaultMaxValue));
+  const isControlled =
+    (minValue !== undefined && minValue !== null) || (maxValue !== undefined && maxValue !== null);
+  const position = isControlled ? init(minValue, maxValue) : state;
+  const setPosition = isControlled ? onChange : setState;
+
+  /*
+   * Effects
+   */
+
+  useEffect(() => {
+    const handleResize = debounce(() => {
+      if (trackRef.current) {
+        setTrackRect(trackRef.current.getBoundingClientRect());
+      }
+    }, 100);
+
+    handleResize();
+
+    const win = doc.defaultView || window;
+
+    win.addEventListener('resize', handleResize);
+
+    return () => {
+      win.removeEventListener('resize', handleResize);
+    };
+  }, [doc, trackRef]);
+
+  /*
+   * Utilities
+   */
+
+  const getTrackPosition = useCallback(
+    (event: MouseEvent) => {
+      let retVal = undefined;
+
+      if (trackRect) {
+        const offset = rtl ? trackRect.left + trackRect.width : trackRect.left;
+        const mouseX = (event.pageX - offset) * (rtl ? -1 : 1);
+        const x = (((max - min) * mouseX) / trackRect.width) as unknown as string;
+        const value = min + parseInt(x, 10);
+
+        // Reduce return value to align with step size.
+        retVal = Math.floor(value / step) * step;
+      }
+
+      return retVal;
+    },
+    [max, min, trackRect, rtl, step]
+  );
+
+  const setThumbPosition = useCallback(
+    (thumb: 'min' | 'max') => (value: number) => {
+      if (!disabled) {
+        let newMinValue = position.minValue;
+        let newMaxValue = position.maxValue;
+
+        if (thumb === 'min') {
+          if (value < min) {
+            newMinValue = min;
+          } else if (value > position.maxValue) {
+            newMinValue = position.maxValue;
+          } else {
+            newMinValue = value;
+          }
+        } else if (thumb === 'max') {
+          if (value > max) {
+            newMaxValue = max;
+          } else if (value < position.minValue) {
+            newMaxValue = position.minValue;
+          } else {
+            newMaxValue = value;
+          }
+        }
+
+        if (newMinValue !== position.minValue || newMaxValue !== position.maxValue) {
+          setPosition({ minValue: newMinValue, maxValue: newMaxValue });
+        }
+      }
+    },
+    [disabled, max, min, position.maxValue, position.minValue, setPosition]
+  );
+
+  /*
+   * Prop getters
+   */
+
+  const getTrackProps = useCallback<IUseSliderReturnValue['getTrackProps']>(
+    ({ onMouseDown, ...other } = {}) => {
```

Comment:
I've noticed that, when spreading objects, we use `...other` in some places and `...props` elsewhere. Just checking/confirming that the preference for this convention is to use `...other`?

Reply:
We try to remain consistent with local codebase conventions – some of which have evolved more than have been documented. In `react-containers` that tends to be `...other`. As in: "the entire parameter is `props`, while the deconstructed one is `x` + `y` + `...other`".

Reply:
@jgorfine-zendesk Curious about the importance of this convention, so we don't end up overloading our contribution docs.

(Note though, that this shouldn't block this PR.)

Reply:
@smritimadan totally agree that documenting this convention shouldn't block the PR. 👍 

I think it's important to do at some point, though. After all, if it's an expectation we have for the code base (and for contributors), then we should communicate it somewhere.